src/Microsoft.DocAsCode.Build.SchemaDriven/Processors/XrefResolverInterpreter.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.SchemaDriven.Processors
+{
+    using System;
+
+    using Newtonsoft.Json.Schema;
+
+    using Microsoft.DocAsCode.Plugins;
+    using System.Linq;
+    using Microsoft.DocAsCode.Common;
+
+    public class XrefResolverInterpreter : IInterpreter
+    {
+        /// <summary>
+        /// Report xrefSpec when
+        /// 1. ContentType = uid is defined => internal xref spec
+        /// Or 2. XrefResolver is defined => external xref spec
+        /// </summary>
+        /// <param name="schema"></param>
+        /// <returns></returns>
+        public bool CanInterpret(BaseSchema schema)
+        {
+            if (schema.Type != JSchemaType.Object)
+            {
+                return false;
+            }
+
+            if (!string.IsNullOrEmpty(schema.XrefResolver))
+            {
+                return schema.XrefResolver.StartsWith("uid");
+            }
+
+            if (schema.Properties == null)
+            {
+                return false;
+            }
+
+            if (schema.Properties.TryGetValue("uid", out var baseSchema))
+            {
+                return baseSchema.ContentType == ContentType.Uid;
+            }
+
+            return false;
+        }
+
+        public object Interpret(BaseSchema schema, object value, IProcessContext context, string path)
+        {
+            if (value == null || !CanInterpret(schema))
+            {
+                return value;
+            }
+
+            var uid = JsonPointer.GetChild(value, "uid") as string;
+            if (string.IsNullOrEmpty(uid))
+            {
+                Logger.LogWarning($"Invalid xrefResolver for {path}: uid is not set.");
+                return value;
+            }
+
+            var xrefSpec = new XRefSpec
+            {
+                Uid = uid
+            };
+
+            // parts to skip uid
+            var parts = schema.XrefResolver?.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries).Skip(1) ?? new string[] { "name" };
```

Comment:
>?? new string[] { "name" } [](start = 118, length = 26)

if parts is empty, should also add "name" to it?

Reply:
no, I would expect it is intentional

---
In reply to: [135673960](https://github.com/dotnet/docfx/pull/2008#discussion_r135673960) [](ancestors = 135673960)